model-optimizer/unit_tests/extensions/back/MarkNodesWithShapeValues_test.py

```diff
@@ -99,3 +99,53 @@ def test_run_with_const_input(self):
         graph_ref = build_graph(nodes_ref, edges)
         (flag, resp) = compare_graphs(graph, graph_ref, 'res', check_op_attrs=True)
         self.assertTrue(flag, resp)
+
+    def test_run_with_solitary_shapeof_in_shape_value_subgraph(self):
+        # in this case MarkNodesWithShapeValues must leave graph unchanged
+        # so reference nodes are exactly the same
+
+        inp_shape_1 = int64_array((1, 3, 100, 100))
+        inp_shape_2 = int64_array((1, 3, 100, 50))  # inp_2 and const will be concatenated to (1, 3, 200, 50)
+        const_shape = int64_array((1, 3, 100, 50))
+
+        nodes = {
+            **regular_op_with_shaped_data('input_1', inp_shape_1, {'op': 'Parameter', 'type': 'Parameter'}),
+            **regular_op_with_shaped_data('input_2', inp_shape_2, {'op': 'Parameter', 'type': 'Parameter',
+                                                                   'returns_shape_value': False}),
+            **shaped_const_with_data('const', const_shape),
+            **regular_op_with_empty_data('concat', {'op': 'Concat', 'type': 'Concat', 'axis': 2,
+                                                    'returns_shape_value': False}),
+            **regular_op_with_empty_data('shapeof', {'op': 'ShapeOf', 'type': 'ShapeOf'}),
+            **regular_op_with_empty_data('reshape', {'op': 'Reshape', 'type': 'Reshape'}),
+            **result('res'),
+        }
+
+        nodes_ref = {
+            **regular_op_with_shaped_data('input_1', inp_shape_1, {'op': 'Parameter', 'type': 'Parameter'}),
+            **regular_op_with_shaped_data('input_2', inp_shape_2, {'op': 'Parameter', 'type': 'Parameter',
+                                                                   'returns_shape_value': False}),
+            **shaped_const_with_data('const', const_shape),
+            **regular_op_with_empty_data('concat', {'op': 'Concat', 'type': 'Concat', 'axis': 2,
+                                                    'returns_shape_value': False}),
+            **regular_op_with_empty_data('shapeof', {'op': 'ShapeOf', 'type': 'ShapeOf'}),
+            **regular_op_with_empty_data('reshape', {'op': 'Reshape', 'type': 'Reshape'}),
+            **result('res'),
+        }
```

Comment:
No need to create a copy of the dictionary. Instead, create 2 graphs in line 144: `graph` and `graph_ref` and remove line 148.

Reply:
agree, fixed